casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -801,17 +801,28 @@ object BlockAPI {
     )
   }
 
-  def getDataAtPar[F[_]: Concurrent: RuntimeManager: BlockStore](
+  def getDataAtPar[F[_]: Concurrent: EngineCell: Log: SafetyOracle: BlockStore](
       blockHash: String,
       par: Par,
       usePreStateHash: Boolean
-  ): F[Seq[Par]] =
-    for {
-      block <- BlockStore[F].getUnsafe(blockHash.unsafeHexToByteString)
-      stateHash = if (usePreStateHash) block.body.state.preStateHash
-      else block.body.state.postStateHash
-      sortedPar <- parSortable.sortMatch[F](par).map(_.term)
-      data      <- RuntimeManager[F].getData(stateHash)(sortedPar)
-    } yield data
+  ): F[(Seq[Par], LightBlockInfo)] = {
```

Comment:
Let `Par` parameter be at the first place, it's the main data required for this method.

Also when Casper is not available we should return (and log) an error as in other methods.

Reply:
Done [here](https://github.com/rchain/rchain/pull/3584/commits/77d380dce27161918dced6a81956c0e6ec0354bb)